crates/uv-resolver/src/lock.rs

```diff
@@ -1291,49 +1323,10 @@ struct GitSource {
     kind: GitSourceKind,
 }
 
-/// An error that occurs when a source string could not be parsed.
-#[derive(Clone, Debug, Eq, PartialEq)]
-enum GitSourceError {
-    InvalidSha,
-    MissingSha,
-}
-
-impl GitSource {
-    /// Extracts a git source reference from the query pairs and the hash
-    /// fragment in the given URL.
-    ///
-    /// This also removes the query pairs and hash fragment from the given
-    /// URL in place.
-    fn from_url(url: &mut Url) -> Result<GitSource, GitSourceError> {
-        let mut kind = GitSourceKind::DefaultBranch;
-        let mut subdirectory = None;
-        for (key, val) in url.query_pairs() {
-            match &*key {
-                "tag" => kind = GitSourceKind::Tag(val.into_owned()),
-                "branch" => kind = GitSourceKind::Branch(val.into_owned()),
-                "rev" => kind = GitSourceKind::Rev(val.into_owned()),
-                "subdirectory" => subdirectory = Some(val.into_owned()),
-                _ => continue,
-            };
-        }
-        let precise = GitSha::from_str(url.fragment().ok_or(GitSourceError::MissingSha)?)
-            .map_err(|_| GitSourceError::InvalidSha)?;
-
-        url.set_query(None);
-        url.set_fragment(None);
-        Ok(GitSource {
-            precise,
-            subdirectory,
-            kind,
-        })
-    }
-}
-
 #[derive(Clone, Debug, Eq, Hash, PartialEq, PartialOrd, Ord, serde::Deserialize)]
 enum GitSourceKind {
     Tag(String),
     Branch(String),
-    Rev(String),
```

Comment:
Because, at least in this change, "rev" is no longer a distinct kind. It is strictly duplicative with the fact that the lock file always records a revision regardless of what the underlying kind is.

Reply:
But how does it record the revision that was _requested_ by the user?

Reply:
Ah I see. I _believe_ that is exactly the thing I am unsure about here. As in, I don't think I grok why we need to keep track of what the user requested specifically in the lock file, since it is presumably in `pyproject.toml`. But yeah, if we do need to keep track of what the user requested in the lock file, specifically for git (but notably not for other things), then we will indeed need to capture some extra state here somehow. Ideally without duplication the revision hash.

Reply:
So if the user has `rev = "refs/foo"` in their requirements, and we lock with that request, and then they re-lock with the same requirements... We want to know that the locked requirement matches `rev = "refs/foo"` _even if_ the commit that maps to that ref has changed (unless the user passed `--upgrade`).

Reply:
OK, so my mistake here is assuming that "revision" and "precise sha1" were equivalent. They aren't, because a revision might be a named reference to something. I'll add this back.